Assets/PatchKit Patcher/Scripts/AppData/Local/Pack1Unarchiver.cs

```diff
@@ -68,35 +82,68 @@ public void Unarchive(CancellationToken cancellationToken)
             DebugLogger.Log("Unpacking " + _metaData.Files.Length + " files...");
             foreach (var file in _metaData.Files)
             {
-                OnUnarchiveProgressChanged(file.Name, file.Type == "regular", entry, _metaData.Files.Length, 0.0);
+                OnUnarchiveProgressChanged(file.Name, file.Type == Pack1Meta.regularFileType, entry, _metaData.Files.Length, 0.0);
 
                 var currentFile = file;
                 var currentEntry = entry;
-                Unpack(file, progress =>
+
+                if (CanUnpack(file))
                 {
-                    OnUnarchiveProgressChanged(currentFile.Name, currentFile.Type == "regular", currentEntry, _metaData.Files.Length, progress);
-                });
+                    Unpack(file, progress =>
+                    {
+                        OnUnarchiveProgressChanged(currentFile.Name, currentFile.Type == Pack1Meta.regularFileType, currentEntry, _metaData.Files.Length, progress);
+                    });
+                }
 
-                OnUnarchiveProgressChanged(file.Name, file.Type == "regular", entry, _metaData.Files.Length, 1.0);
+                OnUnarchiveProgressChanged(file.Name, file.Type == Pack1Meta.regularFileType, entry, _metaData.Files.Length, 1.0);
 
                 entry++;
             }
             DebugLogger.Log("Unpacking finished succesfully!");
         }
 
+        public void UnarchiveSingleFile(Pack1Meta.FileEntry file, CancellationToken cancellationToken)
```

Comment:
Unpacking a single file should have an optional possibility to define a target location that would ignore `destinationDirPath` field. Like `UnarchiveSingleFileAs(file, destFilePath, cancellationToken)`

Reply:
I agree.